packages/block-editor/src/components/block-settings-menu/block-settings-dropdown.js

```diff
@@ -91,7 +91,7 @@ export function BlockSettingsDropdown( {
 			return {
 				firstParentClientId: _firstParentClientId,
 				hasReducedUI: getSettings().hasReducedUI,
-				onlyBlock: 1 === getBlockCount(),
+				onlyBlock: 1 === getBlockCount( _firstParentClientId ),
```

Comment:
Unrelated to this fix, so feel free to ignore my question.

Is there a reason we're using the following code in this component instead of `getBlockRootClientId`?

```js
const parents = getBlockParents( firstBlockClientId );
const _firstParentClientId = parents[ parents.length - 1 ];
```

Reply:
Hm.. I couldn't find any reason by quickly looking at it. At first I thought we were using more info from `parents`. I guess we could do a small follow up for some refactoring - we'd need to check the returned values of the previously used selectors.

Reply:
Hm.. I couldn't find any reason by quickly looking at it. At first I thought we were using more info from `parents`. I guess we could do a small follow up for some refactoring - we'd need to check the returned values of the previously used selectors.

Reply:
I got the same impression, so I decided to ask. Thanks, Nik.

Reply:
I actually made the change in https://github.com/WordPress/gutenberg/pull/44827/commits/e1a5c30edd602e0b9f8fcb6b4e2fef2bc50c08c3. It would be appreciated to review again for that small change :)